File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/ev3lejosv1/controlFlowDecisons.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class NEPOprog {
    private static Configuration brickConfiguration;

    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>();
    private Hal hal = new Hal(brickConfiguration, usedSensors);

    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .build();

            new NEPOprog().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }

    float ___X = 0;

    public void run() throws Exception {
        if (!(0 == ___X)) {
            System.out.println("Assertion failed: " + "pos-0" + 0 + "EQ" + ___X);
        }
        if ( true ) {
            ___X += 1;
        }
        if (!(1 == ___X)) {
            System.out.println("Assertion failed: " + "pos-1" + 1 + "EQ" + ___X);
        }
        if ( false ) {
            ___X += 1000;
        }
        if (!(1 == ___X)) {
            System.out.println("Assertion failed: " + "pos-2" + 1 + "EQ" + ___X);
        }
        if ( true ) {
            if ( true ) {
                ___X += 1;
            }
            ___X += 1;
        }
        if (!(3 == ___X)) {
            System.out.println("Assertion failed: " + "pos-3" + 3 + "EQ" + ___X);
        }
        if ( true ) {
            if ( false ) {
                ___X += 1000;
            }
            ___X += 1;
        }
        if (!(4 == ___X)) {
            System.out.println("Assertion failed: " + "pos-4" + 4 + "EQ" + ___X);
        }
        if ( false ) {
            if ( false ) {
                ___X += 1000;
            }
            ___X += 1000;
        }
        if (!(4 == ___X)) {
            System.out.println("Assertion failed: " + "pos-5" + 4 + "EQ" + ___X);
        }
        if ( false ) {
            if ( true ) {
                ___X += 1000;
            }
            ___X += 1000;
        }
        if (!(4 == ___X)) {
            System.out.println("Assertion failed: " + "pos-6" + 4 + "EQ" + ___X);
        }
        if ( true ) {
            if ( true ) {
                if ( false ) {
                    ___X += 1000;
                }
                ___X += 1;
            }
            ___X += 1;
        }
        if (!(6 == ___X)) {
            System.out.println("Assertion failed: " + "pos-7" + 6 + "EQ" + ___X);
        }
        if ( true ) {
            ___X += 1;
        } else if ( false ) {
            ___X += 1000;
        }
        if (!(7 == ___X)) {
            System.out.println("Assertion failed: " + "pos-8" + 7 + "EQ" + ___X);
        }
        if ( false ) {
            ___X += 1000;
        } else if ( true ) {
            ___X += 1;
        }
        if (!(8 == ___X)) {
            System.out.println("Assertion failed: " + "pos-9" + 8 + "EQ" + ___X);
        }
        if ( true ) {
            ___X += 1;
        } else {
            ___X += 1000;
        }
        if (!(9 == ___X)) {
            System.out.println("Assertion failed: " + "pos-10" + 9 + "EQ" + ___X);
        }
        if ( false ) {
            ___X += 1000;
        } else {
            ___X += 1;
        }
        if (!(10 == ___X)) {
            System.out.println("Assertion failed: " + "pos-11" + 10 + "EQ" + ___X);
        }
        if ( true ) {
            ___X += 1;
        } else if ( true ) {
            ___X += 1000;
        } else {
            ___X += 1000;
        }
        if (!(11 == ___X)) {
            System.out.println("Assertion failed: " + "pos-12" + 11 + "EQ" + ___X);
        }
        if ( false ) {
            ___X += 1000;
        } else if ( false ) {
            ___X += 1000;
        } else {
            ___X += 1;
        }
        if (!(12 == ___X)) {
            System.out.println("Assertion failed: " + "pos-13" + 12 + "EQ" + ___X);
        }
        if ( true ) {
            ___X += 1;
        } else if ( false ) {
            ___X += 1000;
        } else {
            ___X += 1000;
        }
        if (!(14 == ___X)) {
            System.out.println("Assertion failed: " + "pos-14" + 14 + "EQ" + ___X);
        }
        if ( false ) {
            ___X += 1000;
        } else if ( true ) {
            ___X += 1;
        } else {
            ___X += 1000;
        }
        if (!(14 == ___X)) {
            System.out.println("Assertion failed: " + "pos-15" + 14 + "EQ" + ___X);
        }
        System.out.println(( ( 14 == ___X ) ? "Control Flow Test: success" : "Control Flow Test: FAIL" ));
    }
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/microbitv2/controlFlowDecisons.py
import microbit
import random
import math

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

timer1 = microbit.running_time()

___X = 0

def run():
    global timer1, ___X
    if not 0 == ___X:
        print("Assertion failed: ", "pos-0", 0, "EQ", ___X)
    if True:
        ___X += 1
    if not 1 == ___X:
        print("Assertion failed: ", "pos-1", 1, "EQ", ___X)
    if False:
        ___X += 1000
    if not 1 == ___X:
        print("Assertion failed: ", "pos-2", 1, "EQ", ___X)
    if True:
        if True:
            ___X += 1
        ___X += 1
    if not 3 == ___X:
        print("Assertion failed: ", "pos-3", 3, "EQ", ___X)
    if True:
        if False:
            ___X += 1000
        ___X += 1
    if not 4 == ___X:
        print("Assertion failed: ", "pos-4", 4, "EQ", ___X)
    if False:
        if False:
            ___X += 1000
        ___X += 1000
    if not 4 == ___X:
        print("Assertion failed: ", "pos-5", 4, "EQ", ___X)
    if False:
        if True:
            ___X += 1000
        ___X += 1000
    if not 4 == ___X:
        print("Assertion failed: ", "pos-6", 4, "EQ", ___X)
    if True:
        if True:
            if False:
                ___X += 1000
            ___X += 1
        ___X += 1
    if not 6 == ___X:
        print("Assertion failed: ", "pos-7", 6, "EQ", ___X)
    if True:
        ___X += 1
    elif False:
        ___X += 1000
    if not 7 == ___X:
        print("Assertion failed: ", "pos-8", 7, "EQ", ___X)
    if False:
        ___X += 1000
    elif True:
        ___X += 1
    if not 8 == ___X:
        print("Assertion failed: ", "pos-9", 8, "EQ", ___X)
    if True:
        ___X += 1
    else:
        ___X += 1000
    if not 9 == ___X:
        print("Assertion failed: ", "pos-10", 9, "EQ", ___X)
    if False:
        ___X += 1000
    else:
        ___X += 1
    if not 10 == ___X:
        print("Assertion failed: ", "pos-11", 10, "EQ", ___X)
    if True:
        ___X += 1
    elif True:
        ___X += 1000
    else:
        ___X += 1000
    if not 11 == ___X:
        print("Assertion failed: ", "pos-12", 11, "EQ", ___X)
    if False:
        ___X += 1000
    elif False:
        ___X += 1000
    else:
        ___X += 1
    if not 12 == ___X:
        print("Assertion failed: ", "pos-13", 12, "EQ", ___X)
    if True:
        ___X += 1
    elif False:
        ___X += 1000
    else:
        ___X += 1000
    if not 14 == ___X:
        print("Assertion failed: ", "pos-14", 14, "EQ", ___X)
    if False:
        ___X += 1000
    elif True:
        ___X += 1
    else:
        ___X += 1000
    if not 14 == ___X:
        print("Assertion failed: ", "pos-15", 14, "EQ", ___X)
    print("Control Flow Test: success" if ( 14 == ___X ) else "Control Flow Test: FAIL")

def main():
    try:
        run()
    except Exception as e:
        raise

if __name__ == "__main__":
    main()